### PR TITLE
deps: update libp2p to 0.45.x

### PR DESCRIPTION
**Pull request by wemeetagain**

### CODE REVIEW

File: packages/beacon-node/src/constants/network.ts [hunk 1]
```diff
@@ -47,4 +47,6 @@ export const GOODBYE_KNOWN_CODES: Record<string, string> = {
 export enum Libp2pEvent {
   peerConnect = "peer:connect",
```
Comment (twoeths) [inline packages/beacon-node/src/constants/network.ts]: looks like peerConnect is not used anymore? if yes we should remove it here

**Comment by twoeths** [comment]
the upgrade looks good to me, can't review changes in every package but I trust @wemeetagain , dropped a comment
@wemeetagain I deployed this branch to feat2 mainnet node, will approve if metrics are comparable to `unstable`

**Comment by twoeths** [comment]
some issue I see after deploying this on a mainnet node:
- cannot maintain consistent number of mesh peers like in `unstable`
- send and receive too many goodbye requests
- lodestar peer score is not as good as unstable
- high Dial Error rate (which could lead to peer score issue above)
- more disconnect - outbound rate

**Comment by wemeetagain** [comment]
Latest metrics appear comparable to unstable, ready for re-review

**Comment by twoeths** [comment]
metrics look great:
- almost no dropped attestations in the last 6h (vs 60% on the mainnet node)
Screenshot 2023-07-05 at 08 28 00

- almost 3x attestations were forwarded compared to stable (this is close to network thread config)
Screenshot 2023-07-05 at 08 29 04

- time for block to become head, I/O lag is the same to stable

- this could be because lower mesh peers compared to stable
Screenshot 2023-07-05 at 08 33 20

**Review by twoeths — APPROVED** [review]

**Comment by nflaig** [comment]
Getting a bunch of theses errors running unstable now

```
Jul-05 13:35:02.052[]                error: uncaughtException: Cannot read properties of undefined (reading '0')
TypeError: Cannot read properties of undefined (reading '0')
    at Libp2pNode.PeerDiscovery.onDiscoveredPeer (file:///home/devops/goerli/lodestar/packages/beacon-node/src/network/peers/discover.ts:276:60)
    at Libp2pNode.[nodejs.internal.kHybridDispatch] (node:internal/event_target:762:20)
    at Libp2pNode.dispatchEvent (node:internal/event_target:704:26)
    at Libp2pNode.dispatchEvent (file:///home/devops/goerli/lodestar/node_modules/@libp2p/interfaces/src/events.ts:64:26)
    at Libp2pNode.safeDispatchEvent (file:///home/devops/goerli/lodestar/node_modules/@libp2p/interfaces/src/events.ts:79:17)
    at EventEmitter.<anonymous> (file:///home/devops/goerli/lodestar/node_modules/libp2p/src/libp2p.ts:102:14)
    at EventEmitter.[nodejs.internal.kHybridDispatch] (node:internal/event_target:762:20)
    at EventEmitter.dispatchEvent (node:internal/event_target:704:26)
    at EventEmitter.dispatchEvent (file:///home/devops/goerli/lodestar/node_modules/@libp2p/interfaces/src/events.ts:64:26)
    at EventEmitter.Libp2pNode.events.dispatchEvent (file:///home/devops/goerli/lodestar/node_modules/libp2p/src/libp2p.ts:65:30) Cannot read properties of undefined (reading '0'
```

**Review by alokeutpal — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.10.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.10.0) :tada:

**Pull request merged**